python/src/scippneutron/file_loading/_log_data.py

```diff
@@ -59,39 +51,33 @@ def _correct_nxlog_times(raw_times: sc.Variable,
             not provided, defaults to 1 (a no-op scaling factor).
     """
     try:
-        raw_times_s = sc.to_unit(raw_times, sc.units.s)
+        raw_times_ns = sc.to_unit(raw_times, sc.units.ns)
     except sc.UnitError:
         raise BadSource(f"The units of time in the NXlog entry at "
                         f"'{group_path}/time{{units}}' must be convertible to seconds, "
                         f"but this cannot be done for '{raw_times.unit}'. Skipping "
                         f"loading NXLog at '{group_path}'.")
 
-    try:
-        _log_start_ts = sc.scalar(
-            value=parse_date(log_start).timestamp() if log_start is not None else 0.,
-            unit=sc.units.s,
-            dtype=sc.dtype.float64)
-    except (ParserError, OverflowError):
-        raise BadSource(
-            f"The date string '{log_start}' in the NXLog entry at "
-            f"'{group_path}/time@start' failed to parse as an ISO8601 date. "
-            f"Skipping loading NXLog at '{group_path}'")
-
-    try:
-        _run_start_ts = sc.scalar(
-            value=parse_date(run_start).timestamp() if run_start is not None else 0.,
-            unit=sc.units.s,
-            dtype=sc.dtype.float64)
-    except (ParserError, OverflowError):
-        raise BadSource(f"The run start time '{run_start}' at '/<NXEntry>/start_time' "
-                        f"failed to parse as an ISO8601 date. Skipping loading NXLog "
-                        f"at '{group_path}'.")
+    if log_start is not None:
+        try:
+            _start_ts = sc.scalar(value=np.datetime64(log_start),
+                                  unit=sc.units.ns,
+                                  dtype=sc.dtype.datetime64)
+        except ValueError:
+            raise BadSource(
+                f"The date string '{log_start}' in the NXLog entry at "
+                f"'{group_path}/time@start' failed to parse as an ISO8601 date. "
+                f"Skipping loading NXLog at '{group_path}'")
+    else:
+        _start_ts = sc.scalar(value=np.datetime64("1970-01-01T00:00:00Z"),
+                              unit=sc.units.ns,
+                              dtype=sc.dtype.datetime64)
 
     _scale = sc.scalar(value=scaling_factor if scaling_factor is not None else 1.,
                        unit=sc.units.dimensionless,
                        dtype=sc.dtype.float64)
 
-    return (raw_times_s * _scale) + (_log_start_ts - _run_start_ts)
+    return _start_ts + (raw_times_ns * _scale).astype(sc.dtype.int64)
```

Comment:
```suggestion
    return _start_ts + (raw_times_ns * _scale).astype(sc.dtype.int64, copy=False)
```
to avoid redundant copy in case type is already correct.